### PR TITLE
Several fixes

### PR DESCRIPTION
**Pull request by mdblack98**
Fixes Icom SATMODE rig operation with WSJT-X
Fixes some error and warnings from cppcheck

### CODE REVIEW

File: yaesu/ft980.c [hunk 1]
```diff
@@ -251,7 +251,7 @@ const struct rig_caps ft980_caps =
     .set_mem  =           ft980_set_mem,
     .get_mem  =           ft980_get_mem,
 
-#if 0
+#ifdef XXREMOVEDXX
```
Comment (N0NB) [inline yaesu/ft980.c]: Clever!  I like that.

**Comment by N0NB** [comment]
I see this on my desktop:

```
make[1]: Entering directory '/home/nate/git/hlb/elad'
  CC       elad.lo
../../hamlib/elad/elad.c: In function ‘elad_set_ctcss_tone_tn’:
../../hamlib/elad/elad.c:2708:39: warning: ‘%02d’ directive output may be truncated writing between 2 and 10 bytes into a region of size 4 [-Wformat-truncation=]
         snprintf(buf, sizeof(buf), "TN%02d", i + 1);
                                       ^~~~
../../hamlib/elad/elad.c:2708:36: note: directive argument in the range [1, 2147483647]
         snprintf(buf, sizeof(buf), "TN%02d", i + 1);
                                    ^~~~~~~~
../../hamlib/elad/elad.c:2708:9: note: ‘snprintf’ output between 5 and 13 bytes into a destination of size 6
         snprintf(buf, sizeof(buf), "TN%02d", i + 1);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../hamlib/elad/elad.c:2704:41: warning: ‘%02d’ directive output may be truncated writing between 2 and 10 bytes into a region of size 3 [-Wformat-truncation=]
         snprintf(buf, sizeof(buf), "TN%c%02d", c, i + 1);
                                         ^~~~
../../hamlib/elad/elad.c:2704:36: note: directive argument in the range [1, 2147483647]
         snprintf(buf, sizeof(buf), "TN%c%02d", c, i + 1);
                                    ^~~~~~~~~~
../../hamlib/elad/elad.c:2704:9: note: ‘snprintf’ output between 6 and 14 bytes into a destination of size 6
         snprintf(buf, sizeof(buf), "TN%c%02d", c, i + 1);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../hamlib/elad/elad.c: In function ‘elad_set_ctcss_sql’:
../../hamlib/elad/elad.c:2860:39: warning: ‘%02d’ directive output may be truncated writing between 2 and 10 bytes into a region of size 4 [-Wformat-truncation=]
         snprintf(buf, sizeof(buf), "CN%02d", i + 1);
                                       ^~~~
../../hamlib/elad/elad.c:2860:36: note: directive argument in the range [1, 2147483647]
         snprintf(buf, sizeof(buf), "CN%02d", i + 1);
                                    ^~~~~~~~
../../hamlib/elad/elad.c:2860:9: note: ‘snprintf’ output between 5 and 13 bytes into a destination of size 6
         snprintf(buf, sizeof(buf), "CN%02d", i + 1);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../hamlib/elad/elad.c:2856:41: warning: ‘%02d’ directive output may be truncated writing between 2 and 10 bytes into a region of size 3 [-Wformat-truncation=]
         snprintf(buf, sizeof(buf), "CN%c%02d", c, i + 1);
                                         ^~~~
../../hamlib/elad/elad.c:2856:36: note: directive argument in the range [1, 2147483647]
         snprintf(buf, sizeof(buf), "CN%c%02d", c, i + 1);
                                    ^~~~~~~~~~
../../hamlib/elad/elad.c:2856:9: note: ‘snprintf’ output between 6 and 14 bytes into a destination of size 6
         snprintf(buf, sizeof(buf), "CN%c%02d", c, i + 1);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CC       fdm_duo.lo
```

**Comment by N0NB** [comment]
Looks good to me!  Aplied and pushed, thanks, Mike.

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: https://www.n0nb.us
Projects: https://github.com/N0NB
GPG fingerprint: 82D6 4F6B 0E67 CD41 F689 BBA6 FB2C 5130 D55A 8819

**Comment by mdblack98** [comment]
That cppcheck has found a few bugs along with some cleanup.  I'm going through them all.
Debating whether or not to #ifdef out the unused functions but I'm leaning towards doing that.  No need to compile them if nobody is using them but leave them there for posterity.
Mike

 

    On Saturday, December 7, 2019, 07:42:00 AM CST, Nate Bargmann <notifications@github.com> wrote:  
 
 Looks good to me! Aplied and pushed, thanks, Mike.

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds. The pessimist fears this is true."

Web: https://www.n0nb.us
Projects: https://github.com/N0NB
GPG fingerprint: 82D6 4F6B 0E67 CD41 F689 BBA6 FB2C 5130 D55A 8819

**Comment by N0NB** [comment]
* On 2019 07 Dec 07:45 -0600, Michael Black wrote:
> That cppcheck has found a few bugs along with some cleanup.  I'm going
> through them all.  Debating whether or not to #ifdef out the unused
> functions but I'm leaning towards doing that.  No need to compile them
> if nobody is using them but leave them there for posterity.  Mike

I've not looked closely, but are they truly unused or is cppcheck simply
not finding them referenced from another part of the library?

I've also run the clang analyzer through the scan-build utility on
Debian Bullseye (Testing repository) and it seems to have caught some
additional things.  As every tool likely uses a different algorithm, I
figure that checking with a couple might give us a reasonably
clean/proper code base before the 4.0 release.

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: https://www.n0nb.us
Projects: https://github.com/N0NB
GPG fingerprint: 82D6 4F6B 0E67 CD41 F689 BBA6 FB2C 5130 D55A 8819

**Comment by N0NB** [comment]
On 2019-12-07 14:45, Michael Black wrote:
> That cppcheck has found a few bugs along with some cleanup.  I'm going
> through them all.
>  Debating whether or not to #ifdef out the unused functions but I'm
> leaning towards doing that.  No need to compile them if nobody is
> using them but leave them there for posterity.

I had to look up posterty here -- since git keeps all the history -- why 
not
remove the code #ifdef'ing them out with some good commit messages?
I thing it keeps the focus better that way and there still is history.

Roeland

**Comment by N0NB** [comment]
On 2019-12-07 14:41, Nate Bargmann wrote:
> Looks good to me!  Aplied and pushed, thanks, Mike.

and wow Mike, good job there! (building/installing as we speak)

**Comment by mdblack98** [comment]
I'd rather keep that in the code at least until we do a major release....without an explanation as to why it was removed it may simply be "not implemented yet".  Eventually, I may investigate each of these to determine if removal should be permanent.
Mike

 

    On Sunday, December 8, 2019, 06:12:52 AM CST, Nate Bargmann <notifications@github.com> wrote:  
 
 On 2019-12-07 14:45, Michael Black wrote:
> That cppcheck has found a few bugs along with some cleanup. I'm going
> through them all.
> Debating whether or not to #ifdef out the unused functions but I'm
> leaning towards doing that. No need to compile them if nobody is
> using them but leave them there for posterity.

I had to look up posterty here -- since git keeps all the history -- why 
not
remove the code #ifdef'ing them out with some good commit messages?
I thing it keeps the focus better that way and there still is history.

Roeland

**Comment by N0NB** [comment]
* On 2019 08 Dec 06:13 -0600, roeland@linux-it.nl wrote:
> On 2019-12-07 14:45, Michael Black wrote:
> > That cppcheck has found a few bugs along with some cleanup.  I'm going
> > through them all.
> >  Debating whether or not to #ifdef out the unused functions but I'm
> > leaning towards doing that.  No need to compile them if nobody is
> > using them but leave them there for posterity.
> 
> I had to look up posterty here -- since git keeps all the history -- why not
> remove the code #ifdef'ing them out with some good commit messages?
> I thing it keeps the focus better that way and there still is history.

It can work that way for some.  I think in many ways "out of sight, out
of mind" applies and even though the removed functions would be in a
prior commit in Git, still one has to know to look for them which can be
tedious.  As I see it, #ifdef those sections out makes them invisible to
the compiler as the preprocessor simply doesn't write those sections
into its output, as I understand it.  Yeah, it may make the preprocessor
slightly slower, but on modern machines it's probably not enough for us
to notice sitting at the keyboard.

:-)

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: https://www.n0nb.us
Projects: https://github.com/N0NB
GPG fingerprint: 82D6 4F6B 0E67 CD41 F689 BBA6 FB2C 5130 D55A 8819

**Comment by N0NB** [comment]
On 2019-12-08 16:16, Nate Bargmann wrote:
> * On 2019 08 Dec 06:13 -0600, roeland@linux-it.nl wrote:

> It can work that way for some.  I think in many ways "out of sight, out
> of mind" applies and even though the removed functions would be in a
> prior commit in Git, still one has to know to look for them which can 
> be
> tedious.  As I see it, #ifdef those sections out makes them invisible 
> to
> the compiler as the preprocessor simply doesn't write those sections
> into its output, as I understand it.  Yeah, it may make the 
> preprocessor
> slightly slower, but on modern machines it's probably not enough for us
> to notice sitting at the keyboard.

some thoughts


https://stackoverflow.com/questions/15699995/could-someone-explain-the-pros-of-deleting-or-keeping-unused-code/15700228#15700228

I think the fist post nailed it. But I'm not the maintainer ;-)

R.

**Comment by N0NB** [comment]
* On 2019 08 Dec 11:55 -0600, roeland@linux-it.nl wrote:
> On 2019-12-08 16:16, Nate Bargmann wrote:
> > * On 2019 08 Dec 06:13 -0600, roeland@linux-it.nl wrote:
> 
> > It can work that way for some.  I think in many ways "out of sight, out
> > of mind" applies and even though the removed functions would be in a
> > prior commit in Git, still one has to know to look for them which can be
> > tedious.  As I see it, #ifdef those sections out makes them invisible to
> > the compiler as the preprocessor simply doesn't write those sections
> > into its output, as I understand it.  Yeah, it may make the preprocessor
> > slightly slower, but on modern machines it's probably not enough for us
> > to notice sitting at the keyboard.
> 
> some thoughts
> 
> 
> https://stackoverflow.com/questions/15699995/could-someone-explain-the-pros-of-deleting-or-keeping-unused-code/15700228#15700228
> 
> I think the fist post nailed it. But I'm not the maintainer ;-)

I've no real argument against the points made.  I will say that if an
#idef block is large enough that grep turns up a keyword in that area
and the #ifdef and #endif are far enough above and below that neither
shows in the editor, then that is a potential time waster.  We do have
some large chunks of #ifdef'd code, mostly where Windows support and
POSIX differ to a large extent, but, as I recall, those are used blocks
of code and need to remain.

If it's obvious that it hasn't been touched for a long time and those
functions are not called anywhere, then it is safest to remove them with
an announcement to the list some time in advance to be polite.

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: https://www.n0nb.us
Projects: https://github.com/N0NB
GPG fingerprint: 82D6 4F6B 0E67 CD41 F689 BBA6 FB2C 5130 D55A 8819

**Pull request merged**